core/src/main/java/org/apache/accumulo/core/metadata/StoredTabletFile.java

```diff
@@ -291,9 +291,41 @@ public TabletFileCq(Path path, Range range) {
     }
   }
 
-  private static class TabletFileCqMetadataGson {
+  static class TabletFileCqMetadataGson {
     private String path;
     private byte[] startRow;
     private byte[] endRow;
+
+    TabletFileCqMetadataGson() {}
+
+    TabletFileCqMetadataGson(AbstractTabletFile<?> atf) {
+      path = Objects.requireNonNull(atf.path.toString());
+      startRow = encodeRow(atf.range.getStartKey());
+      endRow = encodeRow(atf.range.getEndKey());
+    }
+
+    ReferencedTabletFile toReferencedTabletFile() {
+      return new ReferencedTabletFile(new Path(URI.create(path)),
+          new Range(decodeRow(startRow), true, decodeRow(endRow), false));
+    }
+
+    StoredTabletFile toStoredTabletFile() {
+      return new StoredTabletFile(new TabletFileCq(new Path(URI.create(path)),
+          new Range(decodeRow(startRow), true, decodeRow(endRow), false)));
+    }
+  }
+
+  static StoredTabletFile fomTabletFileCqMetadataGson(String serialized) {
+    return new StoredTabletFile(deserialize(serialized));
+  }
```

Comment:
This method does not seem to be used.

Reply:
Good catch. This was left-over from an initial cut at fixing this. I removed it.